src/errors.ts

```diff
@@ -31,6 +31,16 @@ export class CancelledError extends BaseError {
 }
 
 /**
+ * HTTPError is used when a request does not respond successfully.
+ */
+export class HTTPError extends BaseError {
+    constructor(public code: number, message: string) {
+        super(message);
+        HTTPError.setProto(this);
+    }
```

Comment:
One last commit, I promise! Probably a good idea to capture the raw http.IncomingMessage here for consumers, and expose it via a `cause()` method.